### PR TITLE
feat(scrim): New calcite-scrim component

### PR DESCRIPTION
**Pull request by driskull**
New Component: Calcite-scrim #252

We want a more widespread component for loading and disabled states. Gonna make our calcite-app functional component into a calcite "component component" that can be used in this repo and brought into arcgis-components (for Dhrumil).

Maybe related to loading / skeleton states? #163

### CODE REVIEW

**Comment by paulcpederson** [comment]
This should probably be used in `calcite-modal`, yeah? Or am I misunderstanding what this is for?

**Comment by driskull** [comment]
@paulcpederson I'm not sure. @asangma would you know? Seems like the modal uses a darker background than this but I guess it could.

**Comment by paulcpederson** [comment]
Yeah looks like the dark theme is pretty much what we have for modal, so could maybe do that?

**Comment by driskull** [comment]
I'll wait for @asangma to chime in because I'm not sure if the `calcite-scrim` is intended to be different from the Modal styling since this scrim is primary for panel usage within the map viewer.

To me, it makes sense to be used for any overlay scenario like the Modal, but we'll have to get the styles in sync.

**Comment by paulcpederson** [comment]
niiiice, thanks @driskull

File: src/components/calcite-scrim/calcite-scrim.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+:host {
+  display: flex;
+  position: relative;
+  --calcite-scrim-background: #{rgba($blk-000, 0.75)};
+}
+
+// dark theme
+:host([theme="dark"]) {
+  --calcite-scrim-background: #{rgba($blk-200, 0.75)};
```
Comment (asangma) [inline src/components/calcite-scrim/calcite-scrim.scss:9]: Should this use `$blk-240` which would match what the modal previously had?
Modal was setting the scrim color to `--calcite-modal-scrim: rgba(0, 0, 0, 0.75);`

Comment (driskull) [reply src/components/calcite-scrim/calcite-scrim.scss:9]: Sure. I'll update. What about the light theme color?

Comment (asangma) [reply src/components/calcite-scrim/calcite-scrim.scss]: Looks good! 💥

**Pull request merged**